File: 0155-min-stack/0155-min-stack.js

var MinStack = function() {
    this.items = [];
};

/** 
 * @param {number} val
 * @return {void}
 */
MinStack.prototype.push = function(val) {
    this.items.push(val);
};

/**
 * @return {void}
 */
MinStack.prototype.pop = function() {
    this.items.pop();
};

/**
 * @return {number}
 */
MinStack.prototype.top = function() {
    return this.items[this.items.length - 1]
};

/**
 * @return {number}
 */
MinStack.prototype.getMin = function() {
    return this.items.slice().sort((a, b) => a-b)[0];
};

/** 
 * Your MinStack object will be instantiated and called as such:
 * var obj = new MinStack()
 * obj.push(val)
 * obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.getMin()
 */

File: 0155-min-stack/0155-min-stack.py
class MinStack:

    def __init__(self):
        self.items = []
        self.min_items = []
        
    def push(self, val: int) -> None:
        self.items.append(val)
        if self.min_items:
            val = min(self.min_items[-1], val)
        self.min_items.append(val)
        
    def pop(self) -> None:
        self.items.pop()
        self.min_items.pop()

    def top(self) -> int:
        return self.items[-1]

    def getMin(self) -> int:
        return self.min_items[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()